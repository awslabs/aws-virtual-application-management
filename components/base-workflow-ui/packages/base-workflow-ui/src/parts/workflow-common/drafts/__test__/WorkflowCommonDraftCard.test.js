/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowCommonDraftCard from '../WorkflowCommonDraftCard';

describe('WorkflowCommonDraftCard', () => {
  const onCancel = jest.fn();
  const editor = { version: { selectedSteps: [] }, addStep: jest.fn(), stepEditorsEditing: jest.fn() };
  const stepEditor = {};
  const stepTemplatesStore = { load: jest.fn() };
  const userDisplayName = {
    getDisplayName: jest.fn(() => {
      return 'test-name';
    }),
  };

  beforeEach(() => {
    jest.spyOn(global.Date, 'now').mockImplementation(() => 1626365860332);
  });

  const draft = { template: {}, workflow: {}, createdAt: 'Feb 1, 1966 UTC', createdBy: 'test' };
  it('renders correctly', () =>
    expect(
      renderer
        .create(
          <Provider userDisplayName={userDisplayName} stepTemplatesStore={stepTemplatesStore}>
            <WorkflowCommonDraftCard
              draft={draft}
              editor={editor}
              stepEditor={stepEditor}
              onCancel={onCancel}
              StepTemplatesStore={stepTemplatesStore}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot());
});
