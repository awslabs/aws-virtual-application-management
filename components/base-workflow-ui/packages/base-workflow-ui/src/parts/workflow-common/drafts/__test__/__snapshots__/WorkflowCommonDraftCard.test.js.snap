// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowCommonDraftCard renders correctly 1`] = `
Array [
  <div
    className="ui top left attached label"
    onClick={[Function]}
  >
    Template 
     Draft
  </div>,
  <div
    className=""
  >
    <div
      className="flex"
    >
      <h3
        className="ui grey header mt0 flex-auto ellipsis"
      >
        <div
          className="sub header fs-9 color-grey"
        >
          <div>
            <span
              className="ellipsis breakout"
            />
          </div>
          <div>
            created 
            <time
              dateTime="1966-02-01T00:00:00.000Z"
              title="Feb 1, 1966 UTC"
            >
              55 years ago
            </time>
             
            <span
              className="ml1"
            >
              by 
              test-name
            </span>
          </div>
        </div>
      </h3>
      <div>
        <div
          className="ui mini basic buttons"
        >
          <button
            className="ui icon button"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="edit icon"
              onClick={[Function]}
            />
          </button>
          <button
            className="ui icon button"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="trash icon"
              onClick={[Function]}
            />
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
